### PR TITLE
Desktop two-pane UI with react-navigation v3

### LINKED ISSUE #7399: Use same UI and navigation on desktop and mobile
# Problem

Desktop is currently using a different set of screens and navigation system, which adds up to the burden of maintaining the platform. 

# Implementation

Check what is required to have desktop work with the current mobile screens. We can use conditionals like we already have for ios/android for platform specific things but overall that would remove the only limitation we have for having features released on both platforms simultaneously.

# Acceptance Criteria

Desktop doesn't use it's own screens anymore

# Related issues
- [Improve FlatList performance ](https://github.com/status-im/react-native-desktop/issues/470)
- [Two-pane status UI](https://github.com/status-im/status-react/issues/8168)

# Notes

- react-navigation might need to be implemented for the desktop platform?

### PR DESCRIPTION
**Pull request by vkjr**
fixes #7399

### Summary
This PR contains desktop mobile UI updated to work with react-navigation v3

### Testing notes
Suggestion is to test only mobile platforms.

#### Platforms

- Android
- iOS
- macOS
- Linux
- Windows

##### Functional

- 1-1 chats
- public chats
- group chats
- wallet / transactions
- dapps / app browsing
- account recovery
- new account
- user profile updates
- networks
- mailservers
- fleet
- bootnodes

##### Non-functional

- battery performance
- CPU performance / speed of the app
- network consumption

### Steps to test

- Open Status
- ...
- Step 3, etc.

status: wip

### CODE REVIEW

**Comment by vkjr** [comment]
@annadanchenko, desktop builds in this PR has mobile 2-pane UI working. Can we make a full testing to find all issues that need to be merged before fixing?
So far I see couple:
- Text input in chat gains focus only when clicked in the leftmost part of area. 
- Window that opens after click on "share my profile" has broken layout

**Comment by hesterbruikman** [comment]
Took the UI for a testrun. 

Device
- Windows 10
- 15.6”
- 1920x1080 (16:9)

Some quick observations:

- When opening a new public channel via [+]-button, it's not added to left pane.
- Left pane content might need some scaling down, most channel names become unreadible and all start with Status
- Bottom nav icons might need some scaling up. They are tiny on desktop
- [Select a chat to start messaging should be grey (#939BA1) as it is a hint text, blue is reserved for buttons/links
- Repositioning the cursor on the text entry box in chat is not responsive
- I thought on desktop UI the bttom nav wass intended to be positioned at the bottom of the left pane. Spreading accross the whole bottom only on tablet sizes. @errorists can you confirm?
- [+]-button menu should come up as an overlay or have some shadow at the top
- Notifications probably needs to trigger an OS settings menu
- The icons on the Settings  list items  (i.e. [>] and [switch] are too far on the right. Especially in [Devices] the switch gets cut off. 

IMO this UI is more suitable on a tablet device. Will give it try on 10.1" touch 1920 x 1200 next.

**Comment by vkjr** [comment]
@hesterbruikman, thanks for checking! yes, this UI now is purely mobile, without customization for desktop. I'd be glad if we can merge it in develop after solving technical issues (like Text input in chat) and leave UI notes to be fixed later in a separate PRs. Wdyt, would it be ok?

**Comment by hesterbruikman** [comment]
> 
> 
> @hesterbruikman, thanks for checking! yes, this UI now is purely mobile, without customization for desktop. I'd be glad if we can merge it in develop after solving technical issues (like Text input in chat) and leave UI notes to be fixed later in a separate PRs. Wdyt, would it be ok?

Sure! Let me know where to best capture issues for future PR's.

**Comment by vkjr** [comment]
I think they can be created right here, on github. Maybe new label would be convenient, something like `desktop-mobile`, to not interefere with current `desktop` label which is about old ui.

**Comment by hesterbruikman** [comment]
> 
> 
> I think they can be created right here, on github. Maybe new label would be convenient, something like `desktop-mobile`, to not interefere with current `desktop` label which is about old ui.

Thanks @vkjr! See [here ](https://github.com/status-im/status-react/issues/8661)a list by myself and @errorists. Please check in if anything needs clarification!

**Comment by vkjr** [comment]
Thanks!

**Comment by jakubgs** [comment]
@vkjr 
>Maybe new label would be convenient, something like desktop-mobile, to not interefere with current desktop label which is about old ui.

Labels can be renamed here:
https://github.com/status-im/status-react/labels?utf8=%E2%9C%93&q=desktop
Just click the `Edit` button. We could rename `desktop` to `desktop-old`, and your current `desktop-mobile` to `desktop-new` for clarity.

**Comment by flexsurfer** [comment]
@vkjr i installed mac version but can't see mobile UI :(

**Comment by flexsurfer** [comment]
@vkjr well done, love it so far on mac! :) for some reason it works faster and UI more responsive

**Comment by errorists** [comment]
@flexsurfer r u sure? Weird, myself and some other folks have the exact opposite experience, that it’s more laggy. Maybe there’s some variable in play like Mac OS version or config, dunno.

**Comment by flexsurfer** [comment]
switching between chats and scrolling messages much faster, dunno, maybe you need pro mac for that :D

**Comment by flexsurfer** [comment]
yes after some time it gets slower and slower and more buggy :(

**Comment by annadanchenko** [comment]
@vkjr would you mind to fix issues with builds, please? 
@churik will you have time to test 2-pane UI PR soon?

**Comment by churik** [comment]
@annadanchenko Yes, I have, but it is still WIP - so not sure what exactly should be fixed from https://github.com/status-im/status-react/pull/7551 and what areas should work for now (i.e. chats only or  chats + wallet + browser) and what platforms I can test.
Would be nice to have this info before testing.

**Comment by vkjr** [comment]
@churik, @annadanchenko, I don't know what was a problem with building last commits, I think it is something with a build system, not the code.
Currently, I'm adding necessary functionality on react-native-desktop side. And today is my last day before long (3 weeks) vacation.
So I think for now let this PR stay WIP. It doesn't contain important cljs code, most changes are on react-native-desktop side, so it is not crucial to be committed.
But anyway it would be nice to have it tested from a functional point of view. To do this you can use first successful builds because the latest commits don't affect functionality a lot.

**Comment by vkjr** [comment]
@jakubgs, could we enabled desktop-platforms for this PR? Can't be tested without them. Also there is a build issue in `prep` stage of ios, could you please take a look?

**Comment by jakubgs** [comment]
>could we enabled desktop-platforms for this PR?

Done.

>there is a build issue in prep stage of ios

```
Caused by: hudson.plugins.git.GitException: Command "git clean -fdx" returned status code 1:
stdout: 
stderr:
warning: failed to remove node_modules/is-plain-obj: Permission denied
warning: failed to remove node_modules/plist: Permission denied
...
```
Looks like a permissions issue. I'll just purge those workspaces for now.
```bash
 $ ansible ci-slave-macos -m shell -o -a 'rm -fr /Users/jenkins/workspace/status-react_prs*'
macos-03.ms-eu-dublin.ci.misc | CHANGED | rc=0 | (stdout) 
macos-01.ms-eu-dublin.ci.misc | CHANGED | rc=0 | (stdout) 
macos-02.ms-eu-dublin.ci.misc | CHANGED | rc=0 | (stdout) 
```
Lets see how this one goes:
https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-8630/9/

**Review by yenda — APPROVED** [review]

**Comment by flexsurfer** [comment]
@vkjr all builds are failed , any updates on this PR ?

**Comment by vkjr** [comment]
@flexsurfer, it has issues after rebase. Until yesterday I was trying to figure out why desktop status  can't login, looks like "node.login" signal was never received. 
Yesterday I decided to rebase again to incorporate realm changes. And now build process fails on `react-native-keychain` library. :( Working on this, also requested help from @PombeirP when he has time.

**Comment by pedropombeiro** [comment]
@vkjr I don't see errors regarding `react-native-keychain`. From the failed builds I see:
- Linux:
> WARNING: Use of undeclared Var status-im.react-native.js-dependencies/net-info at line 225 /home/jenkins/workspace/status-react_prs_linux_PR-8630/components/src/status_im/ui/components/react.cljs

- Android:
> TypeError: status_im.react_native.js_dependencies.dismiss_keyboard.call is not a function

Those aren't related to build system and seem quite easy to fix. Am I missing something?

Btw, if you rebase on `origin/develop` I've fixed the Linux issue there.

**Comment by pedropombeiro** [comment]
Took the liberty to rebase and fix the `react-native-desktop-gesture-handler` build issue (was missing the `/desktop` suffix in the path). Linux is building fine for me now.

**Comment by vkjr** [comment]
@pombeirp, thanks! Issue I have not related to those you mentioned. It is about react-native-keychain build. Not sure, maybe it is something mac-specific. Didn’t find the reason yet. Will setup build on other os if won’t find quickly.

**Comment by pedropombeiro** [comment]
OK, I'll try to build on a Mac

**Comment by jakubgs** [comment]
When is this intended to be merged? What's the progress?

**Comment by pedropombeiro** [comment]
We should remove the compiled realm stuff from the AppImage

**Comment by vkjr** [comment]
@jakubgs, in general, this PR has a lot of small issues -  https://github.com/status-im/status-react/issues/8661
But I think it would be great to merge it as soon as it builds and doesn't affect mobile builds. Because anyway we don't have a working version of the desktop. And different UI problems can be solved separately.
Now the only problem with tests stops us from having build, checking it today.

@churik, wdyt as for merging this PR if it builds and doesn't affect mobile builds?

**Comment by vkjr** [comment]
@jakubgs, could you please take a look at the issue on bundle stage of macos build?

**Comment by jakubgs** [comment]
The error is:
```
Status.app: No such file or directory
```
When running:
```
../scripts/sign-macos-pkg.sh Status.app ../deployment/macos/macos-developer-id.keychain-db.gpg
```
Even though `Status.app` was created based on this message:
```
Package ready in /Users/jenkins/workspace/status-react_prs_macos_PR-8630/StatusImPackage/Status.app!
```
But that means that `Status.app` is created inside of `StatusImPackage`, and the `sign-macos-pkg.sh` script is executed from that folder in CI:
https://github.com/status-im/status-react/blob/c1d23148940616d0ed2ad444b6d97254c2551f4c/ci/desktop.groovy#L4
https://github.com/status-im/status-react/blob/c1d23148940616d0ed2ad444b6d97254c2551f4c/ci/desktop.groovy#L89-L97

So the location should not be an issue.

**Comment by pedropombeiro** [comment]
I see that in this PR a new `desktop/resources` folder is introduced, and we already have a `resources/images` folder. Maybe we should make the structure consistent? The suggestion for a structure depends on whether any assets are shared across desktop and mobile platforms. If they aren't, we could rename `/resources/` to `/mobile/resources/`. If there is partial sharing, maybe we could rename `desktop/resources` to `resources/desktop` after moving the current contents of `resources/images` to `resources/images/mobile`. wdyt?

**Comment by jakubgs** [comment]
The folder `Status.app` clearly exists on the `macos-02` host where the CI job ran:
```
administrator@macos-02.ms-eu-dublin.ci.misc:~ % ls -l /Users/jenkins/workspace/status-react_prs_macos_PR-8630/StatusImPackage/          
total 24032
drwxr-xr-x  3 jenkins  jenkins        96 Sep 18 13:42 Status.app
drwxr-xr-x  3 jenkins  jenkins        96 Sep 18 13:40 assets
-rw-r--r--  1 jenkins  jenkins  12304120 Sep 18 13:40 index.desktop.bundle
```

**Comment by pedropombeiro** [comment]
By the way, are all new PNGs being used? For instance I see a `desktop/resources/tiny_tribute_to_talk.png` which doesn't sound like something we're currently using.

**Comment by jakubgs** [comment]
And we can even see in the log it being signed just before the error:
```
### Signing object...
Status.app: signed app bundle with Mach-O thin (x86_64) [im.status.statusim]
```
So why the fuck the call to `codesign` few lines later fails considering it uses the same `$OBJECT`?
https://github.com/status-im/status-react/blob/c1d23148940616d0ed2ad444b6d97254c2551f4c/scripts/sign-macos-pkg.sh#L109-L110

File: desktop/js_files/package.json
Comment (pedropombeiro) [inline desktop/js_files/package.json]: Before being merged, this reference should be changed from `master` to a tag or a commit hash.

**Comment by jakubgs** [comment]
I added `set -x` in a separate branch and ran that, and it seems like it isn't related to `Status.app` only:
```
### Signing object...
+ '[' -d Status.app ']'
+ codesign \
    --sign 'Developer ID Application: STATUS HOLDINGS PTE. LTD. (DTX7Z4U3YA)' \
    --keychain /var/folders/pc/sgwfk7115_7b8dndw3xgz_0r0000gp/T/tmp.1LBSAnUiCY/macos-developer-id.keychain-db \
    --deep --force --verbose=4 Status.app
Status.app: signed app bundle with Mach-O thin (x86_64) [im.status.statusim]
+ ls -l
total 12020
drwxr-xr-x 3 jenkins jenkins       96 Sep 18 14:09 Status.app
drwxr-xr-x 3 jenkins jenkins       96 Sep 18 14:07 assets
-rw-r--r-- 1 jenkins jenkins 12304497 Sep 18 14:07 index.desktop.bundle
+ ls -l Status.app
total 0
drwxr-xr-x 8 jenkins jenkins 256 Sep 18 14:10 Contents
+ echo -e '\n### Verifying signature...'

### Verifying signature...
+ codesign --verify --strict=all --deep --verbose=4 Status.app
--prepared:/Users/jenkins/workspace/status-react/combined/desktop-macos/StatusImPackage/Status.app/Contents/MacOS/reportApp
--validated:/Users/jenkins/workspace/status-react/combined/desktop-macos/StatusImPackage/Status.app/Contents/MacOS/reportApp
Status.app: No such file or directory
```
It seems - based on Pedros suggestion - this might be more related to the contents of the `Status.app` directory.

**Comment by vkjr** [comment]
@PombeirP, I'm not quite sure what should be final structure of images folder.
Right now I see android resources in `status-react/android/app/src/main/res/` and  ios resources in `status-react/ios/StatusIm/Images.xcassets/`. This is why I added icons for desktop inside  `status-react/desktop/resources`.
But yes, there is also `status-react/resources/images` folder. So altogether it is a bit weird :) Anyway, I'm ready to move icons to any folder. Maybe a bit later since right now that doesn't sound like a priority.

**Comment by jakubgs** [comment]
Would it make sense to generalize the `nix/(linux|windows|macos)/base-image/default.nix` to one file that just takes the arguments for all 3 platforms, or would that be needlessly obtuse?

**Comment by pedropombeiro** [comment]
> Would it make sense to generalize the `nix/(linux|windows|macos)/base-image/default.nix` to one file that just takes the arguments for all 3 platforms, or would that be needlessly obtuse?

I think it makes sense. At this point they're too similar to warrant separate expressions. I'll prepare a PR for it.

File: nix/desktop/base-image/default.nix
Comment (jakubgs) [inline nix/desktop/base-image/default.nix]: Wait, isn't `packageSource.pname` wrong? I seems to be set right before `src` as just `pname` or am I missing something?

**Review by jakubgs — APPROVED** [review]

File: nix/desktop/base-image/default.nix
Comment (pedropombeiro) [reply nix/desktop/base-image/default.nix]: Awesome catch! When merging my script from develop into this branch I missed this bit, thanks!

**Comment by pedropombeiro** [comment]
@vkjr do you want to rebase on develop to fix conflicts? If you want I can do it.

**Comment by vkjr** [comment]
@PombeirP, please no :) I will finish current task and rebase it.
To be honest, I feel a bit uncomfortable when someone pushes additional changes to my PR. Especially if it is rebasing or force-pushing :)

**Comment by pedropombeiro** [comment]
Sure, that's why I asked beforehand. But it's a good idea to keep the branch rebased regularly.

**Comment by pedropombeiro** [comment]
@vkjr are you able to run `make release-desktop` on your machine? I get an error while compiling ClojureScript.

UPDATE: Hmm, nevermind, I get it on `develop` as well. Must be something with my machine:

> WARNING: Use of undeclared Var status-im.utils.platform/os at line 9 /home/pedro/src/github.com/status-im/status-react/src/status_im/ui/screens/hardwallet/pin/styles.cljs

**Comment by yenda** [comment]
@PombeirP are you running latest develop? with this commit 3ee52b4fbabfa5574ac5735e07cb3b7a06c6ad13

File: src/status_im/ui/screens/home/styles.cljs
Comment (yenda) [inline src/status_im/ui/screens/home/styles.cljs]: what happens without this?

File: src/status_im/ui/screens/chat/views.cljs
Comment (yenda) [inline src/status_im/ui/screens/chat/views.cljs]: just inline the style here please since it is a one-liner

File: src/status_im/ui/components/icons/vector_icons.cljs
Comment (yenda) [inline src/status_im/ui/components/icons/vector_icons.cljs]: no point making a merge here, just have :source key and make icon-source return the source as value directly in the map

File: nix/desktop/linux/default.nix
Comment (yenda) [inline nix/desktop/linux/default.nix]: if it is commented out does it mean it could be removed?

**Comment by pedropombeiro** [comment]
> @PombeirP are you running latest develop?

yes

File: modules/react-native-status/desktop/rctstatus.cpp
Comment (yenda) [inline modules/react-native-status/desktop/rctstatus.cpp:409]: did you double check that all the new methods are there and the old ones are removed?

File: components/src/status_im/ui/components/react.cljs
Comment (yenda) [inline components/src/status_im/ui/components/react.cljs]: this should be done in js-dependencies

**Review by yenda — DISMISSED** [review]

**Comment by vkjr** [comment]
@PombeirP, I also got this problem several times after rebase. Don't know what is the reason.  @yenda, yes that happens on a code that includes your changes.

**Comment by vkjr** [comment]
@yenda, I built desktop in release after rebasing pr and still have the same error `can't read property 'os' of undefined`, so looks like your fix didn't help :( Will continue on this issue by removing unused styles like we discussed with @flexsurfer 

@jakubgs, macos build still doesn't work :( could you please take a look?

File: src/status_im/ui/screens/home/styles.cljs
Comment (vkjr) [reply src/status_im/ui/screens/home/styles.cljs]: `react-native-desktop` doesn't have an implemented property that restricts `Text` to be one-line, so without this previews of the last message in a chat list are too big if message is long.

**Comment by yenda** [comment]
please cherry-pick this commit d2e225ded739afcfe3518ecfa486cea63f49c97f and tell me if:
- it compiles on problematic repo
- works at runtime

**Comment by vkjr** [comment]
@yenda, I tried it on non-problematic repo and runtime issue has gone!

**Comment by vkjr** [comment]
@yenda, on problematic repo compile-time issues also gone!
Thanks!

Eagerly waiting this commit in develop

**Comment by pedropombeiro** [comment]
@vkjr I tested the latest state of this branch and still have the red screen. Do you have everything pushed to GH?

**Comment by pedropombeiro** [comment]
Oh, Eric's commit wasn't yet in your branch. I've pushed it, it'll just disappear once you rebase it on a develop branch that already contains the commit.

Confirmed the app now shows the UI for me too.

**Comment by vkjr** [comment]
@jakubgs, could you please take a look at macos build?

**Comment by jakubgs** [comment]
I already said what I think the issue is in https://github.com/status-im/status-react/pull/8630#issuecomment-532682332. I don't know how to fix this, I did not create this desktop build process nor do I know how to fix it.

As far as I can tell the error is about the contents of `Status.app` directory rather than about the directory itself. One way to figure out what is the exact error would be to `strace` it, or in case of a Mac `dtruss` it, but it doesn't work because `codesign` binary has `restricted entitlements`:
```
macos-02:StatusImPackage jenkins$ sudo dtruss codesign --verify --strict=all --deep --verbose=4 Status.app

dtrace: system integrity protection is on, some features will not be available
dtrace: failed to execute codesign: dtrace cannot control executables signed with restricted entitlements
```
So that won't work...

Interestingly enough when i remove the `--strict=all` it does work:
```
macos-02:StatusImPackage jenkins$ codesign --verify --deep --verbose=1 Status.app
Status.app: valid on disk
Status.app: satisfies its Designated Requirement
```
But I also tried with `--strict=symlink` and got:
```
macos-02:StatusImPackage jenkins$ codesign --verify --strict=symlinks --deep --verbose=1 Status.app
Status.app: invalid destination for symbolic link in bundle
file modified: /Users/jenkins/workspace/status-react_prs_macos_PR-8630/StatusImPackage/Status.app/Contents/MacOS/node_modules
```
Which pointed me to the issue:
```
macos-02:StatusImPackage jenkins$ ls -l Status.app/Contents/MacOS/node_modules 
lrwxr-xr-x  1 jenkins  jenkins  25 Sep 28 11:23 Status.app/Contents/MacOS/node_modules -> ../Resources/node_modules
macos-02:StatusImPackage jenkins$ ls -l Status.app/Contents/Resources/node_modules
ls: Status.app/Contents/Resources/node_modules: No such file or directory
```
So then I removed it and checked again:
```
macos-02:StatusImPackage jenkins$ codesign --verify --strict=all --deep --verbose=1 Status.app
Status.app: a sealed resource is missing or invalid
file missing: /Users/jenkins/workspace/status-react_prs_macos_PR-8630/StatusImPackage/Status.app/Contents/MacOS/node_modules
```
But it seems to be necessary. So it looks like you have to fix your build so either:
* `Status.app/Contents/MacOS/node_modules` doesn't exist
* `Status.app/Contents/Resources/node_modules` exists

Not sure if that folder is necessary in the first place, don't know much about desktop builds, but if it doesn't then it shouldn't be in the folder before it's signed.

**Comment by vkjr** [comment]
@jakubgs, thanks a lot for a detailed problem description. I got rid of `node_modules` symlink and now everything builds fine!

**Comment by vkjr** [comment]
@yenda, your comments on code were addressed

File: modules/react-native-status/desktop/rctstatus.cpp
Comment (pedropombeiro) [reply modules/react-native-status/desktop/rctstatus.cpp:409]: ping @vkjr

**Review by pedropombeiro — APPROVED** [review]

**Comment by jakubgs** [comment]
Looks like 42961f3a0ea92ddb9c0bfe9fce7129de0b7d883f broke all desktop builds in `develop` causing:
```
Use of undeclared Var status-im.ui.screens.desktop.main.chat.views/incoming-name at line 69 
/home/jenkins/workspace/status-react/combined/desktop-linux/src/status_im/ui/screens/desktop/main/chat/views.cljs
```

**Comment by flexsurfer** [comment]
on macOS

image
image

**Comment by pedropombeiro** [comment]
Hoping this gets merged soon, as @adambabik's Geth 1.9.5 upgrade branch is blocked on the error that Jakub pointed above, so we can't test if the fix to the Windows build worked.

**Comment by pedropombeiro** [comment]
@flexsurfer didn't you say that this issue was fixed in this branch?

```
WARNING: Use of undeclared Var status-im.ui.screens.desktop.main.chat.views/incoming-name at line 69 /home/jenkins/workspace/status-react_prs_linux_PR-8630/src/status_im/ui/screens/desktop/main/chat/views.cljs
```

**Comment by pedropombeiro** [comment]
The builds are failing because Jenkins rebases this branch on `develop` before building. I kicked a manual build which doesn't rebase and they all pass: https://ci.status.im/job/status-react/job/manual/459/

Once @vkjr rebases this branch on `develop` and fixes the build error it should be green again.

**Comment by vkjr** [comment]
@yenda, @PombeirP, I synced `react-native-status` for desktop with mobile version. Couple keycard-related functions are commented out for now, because there is an [issue in `status-go`](https://github.com/status-im/status-go/issues/1624). But that shouldn't block merge.

**Comment by vkjr** [comment]
@flexsurfer, @churik, @PombeirP, all failed e2e tests are the same as in other merged pr. I suggest to merge this PR, wdyt?

File: src/status_im/ui/components/tabbar/styles.cljs
Comment (flexsurfer) [inline src/status_im/ui/components/tabbar/styles.cljs:86]: why this?

File: src/status_im/ui/screens/chat/input/input.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/chat/input/input.cljs:92]: ?

File: src/status_im/ui/components/tabbar/styles.cljs
Comment (vkjr) [reply src/status_im/ui/components/tabbar/styles.cljs:86]: There is no change in ios style here, I just added a copy of this style for desktop (if I correctly understood your question). Changes highlighted incorrectly in github.

File: src/status_im/ui/screens/chat/input/input.cljs
Comment (vkjr) [reply src/status_im/ui/screens/chat/input/input.cljs:92]: I removed commented out brace because it is leftover and it shouldn't be there.

**Review by flexsurfer — APPROVED** [review]

**Comment by churik** [comment]
@vkjr 
I suppose this enables two-pane UIs, which can be used instead of current desktop.
Did you test it (at least chat functionality)?

**Comment by vkjr** [comment]
@churik, yes, I used it locally for chatting

**Comment by vkjr** [comment]
Merged this after discussion with @flexsurfer. All fixes will be made in a separate PRs.

**Comment by churik** [comment]
For now we should describe briefly what was tested and on what platforms (after changing QA process).
Do you have issues created in GH, that should be addressed separately?

**Comment by vkjr** [comment]
@churik, ok, will do next time. For this PR on mac I tested that account can be restored, I can join public chat or private conversation, send and receive messages.

Bunch of issues created here - https://github.com/status-im/status-react/issues/8661
I know few more, will create github issues today.

**Pull request merged**
